mdal/frmts/mdal_ugrid.cpp

```diff
@@ -20,34 +22,71 @@ MDAL::DriverUgrid *MDAL::DriverUgrid::create()
   return new DriverUgrid();
 }
 
-MDAL::CFDimensions MDAL::DriverUgrid::populateDimensions( const NetCDFFile &ncFile )
+std::string MDAL::DriverUgrid::findMeshName( int dimension, bool optional ) const
+{
+  const std::vector<std::string> variables = mNcFile.readArrNames();
+  for ( const std::string &varName : variables )
+  {
+    const std::string cf_role = mNcFile.getAttrStr( varName, "cf_role" );
+    if ( cf_role == "mesh_topology" )
+    {
+      int topology_dimension = mNcFile.getAttrInt( varName, "topology_dimension" );
+      if ( topology_dimension == dimension )
+      {
+        return varName;
```

Comment:
MDAL supports loading only one (first) mesh. We consider extension to allow multiple meshed (1d, 2d, 3d etc), but it would require API change and big changes in QGIS codebase.